src/main/java/redis/clients/jedis/StreamGroupInfo.java

```diff
@@ -0,0 +1,65 @@
+package redis.clients.jedis;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * This class holds information about a stream group
+ * They can be access via getters.
+ * For future purpose there is also {@link #getGroupInfo()}  method
+ * that returns a generic {@code Map} - in case where more info is returned from a server
+ *
+ */
+public class StreamGroupInfo implements Serializable {
+
+  public final static String NAME = "name";
+  public final static String CONSUMERS = "consumers";
+  public final static String PENDING = "pending";
+  public final static String LAST_DELIVERED = "last-delivered-id";
+
+
+  private final String name;
+  private final long consumers;
+  private final long pending;
+  private final String lastDeliveredId;
+  private final Map<String,Object> groupInfo;
+
+  /**
+   * @param map contains key-value pairs with group info
+   *
+   */
+  public StreamGroupInfo(Map<String, Object> map) {
+    if (map!= null && map.size()>0) {
+     groupInfo = map;
+     name = (String) map.get(NAME);
+     consumers = (long) map.get(CONSUMERS);
+     pending = (long) map.get(PENDING);
+     lastDeliveredId = (String) map.get(LAST_DELIVERED);
+
+    } else throw new IllegalArgumentException();
```

Comment:
If we are throwing runtime exception anyway when map is null isn't it better to just let it throw NullPointerException?

Reply:
Done. My initial thought was that it is not nice to see NullPointers